src/Orbit/Graphics/Shader/VertexLayout.h

```diff
@@ -16,25 +16,18 @@
  */
 
 #pragma once
-#include <string>
 #include <vector>
 
 #include "Orbit/Graphics/Graphics.h"
 
 ORB_NAMESPACE_BEGIN
 
-struct VertexComponent
+enum class VertexComponent : uint8_t
 {
-	enum Type
-	{
-		Float,
-		Vec2,
-		Vec3,
-		Vec4,
-	};
-
-	std::string semantic_name;
-	Type        type;
+	Position,
+	Normal,
+	Color,
+	TexCoord,
 };
 
 using VertexLayout = std::vector< VertexComponent >;
```

Comment:
Consider using std::set, since having multiple of any component is unwanted behavior.